pkgs/by-name/bu/bun/hooks/bun-config-hook.sh

```diff
@@ -0,0 +1,50 @@
+# shellcheck shell=bash
+
+bunConfigHook() {
+    echo "Executing bunConfigHook"
+
+    if [ -n "${bunRoot-}" ]; then
+      pushd "$bunRoot"
+    fi
+
+    if [ -z "${bunDeps-}" ]; then
+      echo "Error: 'bunDeps' must be set when using bunConfigHook."
+      exit 1
+    fi
+
+    echo "Configuring bun store"
+
+    export HOME=$(mktemp -d)
+    export BUN_INSTALL_CACHE_DIR=$(mktemp -d)
```

Comment:
```suggestion
    HOME=$(mktemp -d)
    BUN_INSTALL_CACHE_DIR=$(mktemp -d)
    export HOME BUN_INSTALL_CACHE_DIR
```
https://www.shellcheck.net/wiki/SC2155